fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -937,5 +939,86 @@ void testRecordUpdateReducedMergeForcingAgileTimeQuota() {
         assertTrue(newLength < oldLength);
         assertTrue(timeCommitCount > 0);
     }
+
+    @SuppressWarnings("checkstyle:VariableDeclarationUsageDistance")
+    @ParameterizedTest
+    @BooleanSource
+    void luceneOnlineIndexingByIndexTest(boolean reverseScan) {
+        int numRecords = 47;
+        int transactionLimit = 10;
+        int groupingCount = 1;
+        final Random rn = new Random();
+        rn.nextInt();
+        Index srcIndex = new Index("src_index", field("doc_id"), EmptyKeyExpression.EMPTY, IndexTypes.VALUE, IndexOptions.UNIQUE_OPTIONS);
+        Index tgtIndex = new Index(
+                "Map_with_auto_complete$entry-value",
+                new GroupingKeyExpression(field("entry",
+                        KeyExpression.FanType.FanOut).nest(concat(LuceneIndexTest.keys)), groupingCount + 1),
+                LuceneIndexTypes.LUCENE,
+                ImmutableMap.of());
+
+        long[] docIds = new long[numRecords * 2];
+        for (int i = 0; i < numRecords * 2; i ++) {
+            docIds[i] = rn.nextLong();
```

Comment:
Of note: this could generate the same docId multiple times, which would probably affect the test. I suspect that the odds are low enough that it's nothing to be super concerned with.